src/SimpleSAML/IdP.php

```diff
@@ -317,7 +317,11 @@ public static function postAuth(array $state): void
 
         $idpMetadata = $idp->getConfig()->toArray();
 
-        $pc = new Auth\ProcessingChain($idpMetadata, $spMetadata, 'idp');
+        // Add in authprocs defined in authsources.php
+        $asArray = Configuration::getConfig('authsources.php')->getArray($idp->authSource->getAuthSource()->getAuthId());
+        $asAuthproc = array_key_exists('authproc', $asArray) ? ['authproc' => $asArray['authproc']] : null;
```

Comment:
So my IdP's authsource is an SP.  Won't this change result in the authsource authprocs running twice? once when the SP authsource is used for login, and a second time when run inside of the idp.   That would be a big surprise to people, especially since not all filters can be run in both places.

A second reason doing this "somewhere else" may be the desired location is that the IdP class is just one way of integrating with apps. There is an OIDC module, a casserver module and adfs module.  One challenge in enhancing only the SAML IdP side is when people go to use the other modules they get a different behavior. e.g. you add an OIDC app and now the `authproc` filters configured this new way don't run, even though the authsource is the same.

Reply:
The second point I understand and agree with. It must apply to all backends, not just SAML - I hadn't considered ADFS or OIDC. Thanks for the feedback.

WRT the first point, I think I understand what you mean - the case where we're bridging (https://simplesamlphp.org/docs/stable/simplesamlphp-advancedfeatures.html#bridging-between-protocols)? In that case we're running as both an SP and an IdP, and the "authproc" could be run on the SP side then re-run on the IdP side. Perhaps adding an extra key `authproc_idp` might be acceptable to avoid that clash but still allow IdP authprocs in `authsources.php`?